src/engine/routing_algorithms/alternative_path_mld.cpp

```diff
@@ -76,6 +81,94 @@ struct WeightedViaNodeUnpackedPath
     UnpackedEdges edges;
 };
 
+// Scale the maximum allowed weight increase based on its magnitude:
+//  - Shortest path 10 minutes, alternative 13 minutes => Factor of 0.30 ok
+//  - Shortest path 10 hours, alternative 13 hours     => Factor of 0.30 unreasonable
+double getLongerByFactorBasedOnDuration(const EdgeWeight duration)
+{
+    BOOST_ASSERT(duration != INVALID_EDGE_WEIGHT);
+
+    // We only have generic weights here and no durations without unpacking.
+    // We also have restricted way penalties which are huge and will screw scaling here.
+    //
+    // Users can pass us generic weights not based on durations; we can't do anything about
+    // it here other than either generating too many or no alternatives in these cases.
+    //
+    // We scale the weights with a step function based on some rough guestimates, so that
+    // they match tens of minutes, in the low hours, tens of hours, etc.
+
+    // Computed using a least-squares curve fitiing on the following values:
+    //
+    // def func(xs, a, b, c, d):
+    //       return a + b/(xs-d) + c/(xs-d)**3
+    //
+    // xs = np.array([5 * 60, 10 * 60, 30 * 60, 60 * 60, 3 * 60 * 60, 10 * 60 * 60])
+    // ys = np.array([1.0,    0.75,    0.5,     0.3,     0.2,          0.1])
+    //
+    // xs_interp = np.arange(5*60, 10*60*60, 5*60)
+    // ys_interp = np.interp(xs_interp, xs, ys)
+    //
+    // params, _ = scipy.optimize.curve_fit(func, xs_interp, ys_interp)
+    const constexpr auto a = 9.49571282e-02;
+    const constexpr auto b = 1.25440191e+03;
+    const constexpr auto c = 2.06152165e+09;
+    const constexpr auto d = -1.71666881e+03;
+
+    if (duration < EdgeWeight(5 * 60))
+    {
+        return 1.0;
+    }
+    else if (duration > EdgeWeight(10 * 60 * 60))
+    {
+        return 0.10;
+    }
+
+    // Bigger than 10 minutes but smaller than 10 hours
+    BOOST_ASSERT(duration > 5 * 60 && duration < 10 * 60 * 60);
+
+    return a + b / (duration - d) + c / std::pow(duration - d, 3);
```

Comment:
Ah, so we special case "short" and "long" routes and in between we simply scale the factor?

Reply:
Yeah this is just to bound the values to the range for which we provided samples of durations/parameters to the curve fitting. Outside of that range you don't really know how well this will approximate your intended behavior so we just clamp it.